src/c_api.cpp

```diff
@@ -483,22 +504,39 @@ int LGBM_DatasetCreateFromMat(const void* data,
     omp_set_num_threads(config.num_threads);
   }
   std::unique_ptr<Dataset> ret;
-  auto get_row_fun = RowFunctionFromDenseMatric(data, nrow, ncol, data_type, is_row_major);
+  int32_t total_nrow = 0;
+  for (int j = 0; j < nmat; j++) {
```

Comment:
It seems that LightGBM uses `++j` (prefix) style, but it's better to ask @guolinke.

Reply:
yeah, ++j style